src/main/java/ch/hevs/cloudio/endpoint/CloudioEndpoint.java

```diff
@@ -523,6 +563,138 @@ public InternalEndpoint(String uuidOrAppName, CloudioEndpointConfiguration confi
                 }
             }
 
+            // Is the authentication done with a Token?
+            if (configuration.containsKey(AUTH_TOKEN)) {
+                String token = configuration.getProperty(AUTH_TOKEN, "");
+                String tokenPropertiesPath = configuration.getProperty(AUTH_TOKEN_PROPERTIES_PATH, TOKEN_PROPERTIES_DEFAULT_PATH);
+
+                File tokenPropertiesFile = new File(tokenPropertiesPath.replace("/", File.separator) + File.separator + TOKEN_PROPERTIES_FILENAME);
+
+                if (tokenPropertiesFile.exists() && !tokenPropertiesFile.isDirectory()) {
+                    //nothing to do here, file exist an is populated
+                } else {
+                    //create subfolder for token files
+                    File tokenPropertiesFolder = new File(tokenPropertiesPath);
+                    tokenPropertiesFolder.mkdirs();
+
+                    try {
+
+                        String mainURL = configuration.getProperty(API_URL, null);
+
+                        if (mainURL == null) {
+                            throw new InvalidPropertyException("CloudioEndpoint properties missing: No " + API_URL +
+                                    " properties given as argument");
+                        }
+
+                        HttpRequest request = HttpRequest.newBuilder()
```

Comment:
`HttpRequest request = HttpRequest.newBuilder()
                                .uri(new URI(mainURL + "/api/v1/provision/" + token))
                                .GET()
                                .header("Accept", "application/java-archive")
                                .timeout(Duration.of(10, ChronoUnit.SECONDS))
                                .build();`

New API will use the Accept header  to select the format.